### ISSUE TITLE
Push some small amenities to higher zoom level

### ISSUE THREAD

**Issue by kocio-pl** [gravitystorm/openstreetmap-carto#1745]
I think such small amenities as amenity=post_box and amenity=atm are rendered too early now (z17, like shops for example). They are more important than trash cans (z19), so z18 seems like a natural choice for me.

This issue is related to https://github.com/gravitystorm/openstreetmap-carto/issues/847#issuecomment-127776632 - I have proposed rendering amenity=recycling+recycling_type=container and amenity=waste_disposal also from z18, as they are also only of local interest, but more important than trash cans.

**Comment by HolgerJeromin** [comment]
Is getting the money less important than spending it? I am not sure.

**Comment by kocio-pl** [comment]
I think about proportions: the counterpart for shops is rather bank or post office IMO. ATM or post box are less important than them.

Similar case: amenity=recycling+recycling_type=centre are bigger/more important > amenity=recycling+recycling_type=container and amenity=waste_disposal are in between > trash cans are the least important.

**Comment by kocio-pl** [comment]
Mockup with atm, post_box, telephone [pushed from z17 to z18](https://github.com/kocio-pl/openstreetmap-carto/tree/small-amenities) (and emergency_phone to z19, as this is highly specialized kind of amenity):

z17:
![small-amenities-17](https://cloud.githubusercontent.com/assets/5439713/9305960/99c486fe-44f5-11e5-88b9-2364ee952f49.png)
z18:
![small-amenities-18](https://cloud.githubusercontent.com/assets/5439713/9305962/9c3372e2-44f5-11e5-9336-bd89b8d5d956.png)

I also think about moving them to the low priority group.

Another amenities that could be probably moved further down the zoom level and/or prioritized as less important could be toilettes and drinking_water, but I am not sure about them yet.

**Comment by pnorman** [comment]
Given the density of post boxes, I don't think they should be moved from z17 to z18.

**Comment by kocio-pl** [comment]
It's not about density of post boxes alone - they just add up to the whole density, as you can see, like other small amenities. They are clearly less important than post offices, so in my opinion it's perfectly legit to push them +1 zoom level.

In many places they are of less interest for general public (it's not a map mainly for tourists anyway) and there is [a special issue about objects in rural areas](https://github.com/gravitystorm/openstreetmap-carto/issues/1705), where every POI is more important than in the city/town/village, so post boxes should be more visible there (maybe even more than now - we'll see).

**Comment by HolgerJeromin** [comment]
I think pnorman meant there are only postboxes every 1000 meter (in Germany).

**Comment by kocio-pl** [comment]
Yes, I have understood it this way too - does it sound like I haven't? In other words: I realize that their density is low, but there are other POIs on the map and the sum of all the elements can be dense (as on the example picture) and it makes sense to clear the things (even a bit) if we can. Those small elements tend to be placed around some more important ones, so they don't just take unused space, they rather make busy places even more busy.

BTW: do we have levels of importance or just normal/low groups? I think it would be the best to have more layers - post box may be less important than post office or a shop, but is more important than trash can.

**Comment by math1985** [comment]
> I think such small amenities as amenity=post_box and amenity=atm are rendered too early now (z17, like shops for example). They are more important than trash cans (z19), so z18 seems like a natural choice for me.


+1

**Comment by kocio-pl** [comment]
Detailed technical question: how should I create amenity-medium-priority group? I see that in project.yaml group with id `amenity-low-priority` has something like:

```YAML
ORDER BY prio
          ) AS amenity_low_priority
```

but I don't understand how it is used then to find the real priority level? 

Maybe we don't need additional group, because this line:

```YAML
CASE WHEN amenity='waste_basket' THEN 2 ELSE 1 END AS prio
```

is the core information (the more number, the lower priority) and I could give 3 to waste_basket, 2 for current low priority objects and 1 for those small amenities?

Of course I understand I need to also edit amenity-points.mss file.

**Comment by pnorman** [comment]
> Detailed technical question: how should I create amenity-medium-priority group? I see that in project.yaml group with id `amenity-low-priority` has something like:

> ```YAML
ORDER BY prio
          ) AS amenity_low_priority
```

The name of the table in the SQL (`amenity_low_priority`) exists purely for ease of debugging SQL statements. It could be changed to anything and all that would happen is identifying which layer a query is from when just looking at a SQL log would become very difficult.

The id `amenity-low-priority` is significant, but is just a string of text. You could change it to anything, but would need to change the cartocss the same way.

The "priority" is determined by rendering order, i.e. where the layer is relative to other layers. Within the layer, the order rows are returned in is what determines rendering order, as well as cartocss order.

**Comment by matkoniecz** [comment]
> is the core information (the more number, the lower priority) and I could give 3 to waste_basket, 2 for current low priority objects and 1 for those small amenities?

Yes.

**Comment by kocio-pl** [comment]
I'm waiting for https://github.com/gravitystorm/openstreetmap-carto/pull/1825 to be applied, but [the code is more or less ready](https://github.com/kocio-pl/openstreetmap-carto/tree/small-amenities) and it's working as expected: global density change is not visible, but in some places it's getting more clear and the overall view is more uniform (smaller amenities don't mix with standard ones on z17 and ATMs should not eclipse banks on any zoom level).

**Comment by kocio-pl** [comment]
Another candidate for moving deeper: barrier=bollard should be rendered on z>=19 in my opinion. It also means some other small amenities could be rendered there:
- fire hydrants (simply as a red dot)
- traffic sign poles (silver dot) 
- street lamps (probably some lamp shape in black)

They are also used to be a part of bollard line.

**Comment by maraf24** [comment]
barrier=bollard is very important kind of barrier that makes road impassable for cars. It should be rendered earlier not later. However, there are some bollards which are not part of the road. These could be rendered at z19.

**Comment by kocio-pl** [comment]
Thanks for the comment, it makes a lot of sense! Is there a way to detect in this style if the bollard is part of the road?

**Comment by matkoniecz** [comment]
@kocio-pl 

It should be possible. See turning circles - I think that currently only ones on highway ways are displayed.

**Comment by kocio-pl** [comment]
> It should be rendered earlier not later. 

Currently it is visible from z16 I guess, which zoom level would you like to have as the starting one?

**Comment by BalooUriza** [comment]
http://www.openstreetmap.org/#map=19/33.05398/-96.69600 seems to show that
bollards are rendered, though rather oddly (see the gap between 75 HOV and
75 on the link there; context is the Central Expressway in metro Dallas,
Texas.

On Sun, Sep 13, 2015 at 1:19 PM, Mateusz Konieczny <notifications@github.com

**Comment by kocio-pl** [comment]
I guess highway=traffic_signals are also smaller amenities, so they should be moved to z>=18 instead of z>=17 (which is better for standard amenities).

**Comment by kocio-pl** [comment]
Now I think traffic_signals should be moved to z>=19, since they are not the even the mall amenities worth looking for, rather part of a road infrastructure.

**Comment by matkoniecz** [comment]
Note that some ideas were declined in #1884.

**Comment by kdudzik** [comment]
@kocio-pl would you consider returning to this issue? Maybe #1884 could be split in smaller tasks. I'd be very interested in moving bollards (the ones that don't belong to a highway) to z19 as they tend to mess up the map at z16-z17:
http://www.openstreetmap.org/#map=16/52.2239/20.9200
![image](https://cloud.githubusercontent.com/assets/170668/21205902/fcaefbe0-c25e-11e6-9768-854ac62c6937.png)

**Comment by kocio-pl** [comment]
> would you consider returning to this issue? Maybe #1884 could be split in smaller tasks.

Feel free to do it yourself. I can join the party, but I won't do it alone.

**Comment by kocio-pl** [comment]
I would like to get back to this issue - the summary of a rejected PR was:

>     against moving toilets - 1
>     against moving drinking_water - 2
>     against moving emergency_phone - 1
>     against moving traffic_signals - 2
> 
> The rest (atm, post_box and telephone) was not mentioned directly, however @imagico propositions is completely different and is hard to count.

Would it make sense to create separate PRs for these 3 types to be rendered from z18?

I still don't know how to push bollards to z19+ only if they are not part of highway, but I think this should not create controversy. Can anybody help me with this?

**Comment by kocio-pl** [comment]
Just a reminder to myself: there's a separate issue for this and hint how to do it (#2756).

**Comment by Tomasz-W** [comment]
Just for sort things out - there are 4 tags left in this issue:
* amenity=drinking_water
* amenity=atm
* amenity=post_box
* amenity=telephone

**Comment by Tomasz-W** [comment]
@kocio-pl @Adamant36 
Current state:
- [x] amenity=atm -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3372
- [ ] amenity=post_box -> PR https://github.com/gravitystorm/openstreetmap-carto/issues/3431
- [ ] amenity=drinking_water
- [ ] amenity=telephone

**Comment by Adamant36** [comment]
Thanks. Amenity=post_box is currently an issue though, not a PR. Although, I guess I could have done one for it.

**Comment by Tomasz-W** [comment]
@Adamant36 Oh, I see now. I think that splitting PRs into smaller parts is OK, but I find splitting issues (at least this one) unnecessary for the reason given in https://github.com/gravitystorm/openstreetmap-carto/issues/3298#issuecomment-427260884 .

**Comment by Adamant36** [comment]
@tomasz-W, Good call, even if I disagree. I took the original comment by @kocio-pl about it as a possible suggestion and not a part of the issue. Since he didn't include it in the ATM PR. I guess that's part of the problem with not having unified way of doing things. Issues, lists in issues (sometimes referencing other issues, with their own lists), PRs, lists of PRs, PRs with lists of issues, etc, etc. It can easily turn into inception sometimes ;)

**Comment by Tomasz-W** [comment]
@Adamant36 
Update:
- [x] amenity=atm -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3372
- [x] amenity=post_box https://github.com/gravitystorm/openstreetmap-carto/issues/3431 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3445
- [ ] amenity=drinking_water
- [ ] amenity=telephone

**Comment by Tomasz-W** [comment]
I would add `amenity=charging_station` `historic=wayside_shrine` to the list as they are small and not-so-important objects currently rendered from z17. @kocio-pl What do you think?

**Comment by matkoniecz** [comment]
> amenity=charging_station

Is it equivalent of `amenity=fuel` or is there some major differences in how this objects work?

> historic=wayside_shrine

Many of them appear in rural areas, can you give example of place where they cause problems?

**Comment by jeisenbe** [comment]
Are wayside shrines found in urban areas in Europe? If they are mainly
rural, it would be good to be able to see them at z17

**Comment by dieterdreist** [comment]
Am Mo., 22. Okt. 2018 um 19:05 Uhr schrieb jeisenbe <
notifications@github.com>:

> Are wayside shrines found in urban areas in Europe? If they are mainly
> rural, it would be good to be able to see them at z17



Italy is full of them, in villages, towns, cities, in the country side, at
least we map things like these as wayside shrines:
https://www.fisheaters.com/images/roadsideshrine-sicily.jpg

I estimate we have only mapped a tiny fraction of them.

Cheers,
Martin

**Comment by kocio-pl** [comment]
We had a notable example of this in Cracow:

https://www.openstreetmap.org/#map=18/50.05670/19.93780

While I'm not sure that they are really wayside shrines (I guess most of them are just shrines and should be tagged some other way), there are multiple others that were once near a rural road and now they are inside the urban area, like this one:

https://www.openstreetmap.org/node/5072410155

**Comment by Tomasz-W** [comment]
> Is it equivalent of amenity=fuel or is there some major differences in size/importance?

Yes, due to stats (https://en.wikipedia.org/wiki/Electric_car_use_by_country) electric cars are less than 1% of motor vehicles worldwide, what makes `amenity=charging_station` more a 'fun fact' than a equivalent of `amenity=fuel` at the moment.

Here are some exaples of wayside shrines looking much more important (on the same level as e.g. museums), than they actually are:
https://www.openstreetmap.org/node/4281335169#map=17/52.17473/21.04473
https://www.openstreetmap.org/node/4383580219#map=17/52.40529/16.93988

I think z18 would be better zoom level for both of them.

**Comment by kocio-pl** [comment]
I wanted them to make the balance between being visible in the outdoor and not cluttering the urban areas, but if we are more strict about the size, than z18+ sounds good to me.

**Comment by Adamant36** [comment]
None of the charging stations where I live ever get used by anyone. Except the homeless people to charge their cell phones once in awhile. So I agree with @Tomasz-W that they should be knocked down a zoom level or two.

**Comment by Adamant36** [comment]
@Tomasz-W, id like to revisit this with drinking water and picnic tables. They are causing issues getting in the way of other things where I like to map. Do you have any opinion on either one. I see @kocio-pl mentioned drinking water as a possible option, but I didn't notice anywhere if you are for or against rendering it later.

**Comment by kocio-pl** [comment]
Picnic tables might be just a part of picnic site, and now they are rendered one step later (z16+ and z17+), but z18+ also makes sense for me, because they are small.

We're not able to render outdoor in a proper way until we solve the problem how to recognize them (see #1957), so I don't consider OSM Carto to be usable for such purposes - there are different styles using OSM data, much better suited for this. In the outdoor **any** object is more interesting, simply because there are much less of them, including drinking water. But in general they are very small and sometimes it's a small building with taps, so both z19+ and z18+ makes sense for me.
